src/Microsoft.TemplateEngine.Cli/TemplateInvocationAndAcquisitionCoordinator.cs

```diff
@@ -21,71 +15,49 @@ internal class TemplateInvocationAndAcquisitionCoordinator
         private readonly SettingsLoader _settingsLoader;
         private readonly IEngineEnvironmentSettings _environment;
         private readonly INewCommandInput _commandInput;
-        private readonly TemplateCreator _templateCreator;
-        private readonly IHostSpecificDataLoader _hostDataLoader;
         private readonly ITelemetryLogger _telemetryLogger;
-        private readonly string _defaultLanguage;
         private readonly string _commandName;
         private readonly Func<string> _inputGetter;
         private readonly New3Callbacks _callbacks;
 
-        private bool _resolutionResultInitialized = false;
-        TemplateResolutionResult _templateResolutionResult;
-        ITemplateMatchInfo _templateToInvoke;
-        Status _singleMatchStatus;
-
-        public TemplateInvocationAndAcquisitionCoordinator(SettingsLoader settingsLoader, INewCommandInput commandInput, TemplateCreator templateCreator, IHostSpecificDataLoader hostDataLoader, ITelemetryLogger telemetryLogger, string defaultLanguage, string commandName, Func<string> inputGetter, New3Callbacks callbacks)
+        public TemplateInvocationAndAcquisitionCoordinator(SettingsLoader settingsLoader, INewCommandInput commandInput, ITelemetryLogger telemetryLogger,  string commandName, Func<string> inputGetter, New3Callbacks callbacks)
         {
             _settingsLoader = settingsLoader;
             _environment = _settingsLoader.EnvironmentSettings;
             _commandInput = commandInput;
-            _templateCreator = templateCreator;
-            _hostDataLoader = hostDataLoader;
             _telemetryLogger = telemetryLogger;
-            _defaultLanguage = defaultLanguage;
             _commandName = commandName;
             _inputGetter = inputGetter;
             _callbacks = callbacks;
         }
 
-        public async Task<CreationResultStatus> CoordinateInvocationOrAcquisitionAsync()
+        public async Task<CreationResultStatus> CoordinateInvocationOrAcquisitionAsync(ITemplateMatchInfo templateToInvoke)
```

Comment:
consider supporting `CancellationToken` instances as an optional parameter; then passing it down the calls (if such methods support it).

Reply:
https://github.com/dotnet/templating/issues/2716